client/app/scripts/charts/nodes-chart.js

```diff
@@ -3,17 +3,17 @@ import d3 from 'd3';
 import debug from 'debug';
 import React from 'react';
 import { connect } from 'react-redux';
-import { Map as makeMap, fromJS, is as isDeepEqual } from 'immutable';
+import { Map as makeMap, fromJS } from 'immutable';
 import timely from 'timely';
 
+import { nodeAdjacenciesSelector, adjacentNodesSelector } from '../selectors/chartSelectors';
 import { clickBackground } from '../actions/app-actions';
 import { EDGE_ID_SEPARATOR } from '../constants/naming';
 import { MIN_NODE_SIZE, DETAILS_PANEL_WIDTH, MAX_NODE_SIZE } from '../constants/styles';
 import Logo from '../components/logo';
 import { doLayout } from './nodes-layout';
 import NodesChartElements from './nodes-chart-elements';
-import { getActiveTopologyOptions, getAdjacentNodes,
-  isSameTopology } from '../utils/topology-utils';
+import { getActiveTopologyOptions } from '../utils/topology-utils';
```

Comment:
Yep! This first pass I didn't selectorify much stuff, lots of things can be selector'd!